### PR TITLE
Add changelog for 1.4.0

### PR DESCRIPTION
**Pull request by martinclaus**
I have kept PRs that were removed later on to give credit to the contributors.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,54 @@
 # Changelog
 
+## [1.4]
+
+### [1.4.0] - 2023-03-01
+
+The main purpose of this release is to remove the broken implementation of LTI 1.3 introduced in the previous release.
```
Comment (consideRatio) [inline CHANGELOG.md:7]:
```suggestion
The main purpose of this release is to remove the non-functional `LTI13OAuthenticator` class introduced in 1.3.0 while its being worked on towards a functional state.
```

**Review by consideRatio — APPROVED** [review]
Besides a tweak to help convey that this is to planned to be re-introduced, this looks great to me! Excellent work aligning with practices!!!

Typically I'd merge this PR rather than just approve it, but since its a changelog PR the practice in the jupyterhub team is to often let the author of the changelog PR merge just before making a release according to https://github.com/jupyterhub/ltiauthenticator/blob/main/RELEASE.md.

If you think you are ready, you can do:

1. Resolve my review comment one way or the other
2. Merge
3. Follow RELEASE.md instructions of using `tbump` to push a git tag which will trigger a publication to PyPI via the github workflow's automation.

Thank you @martinclaus for your work on this!!!!

**Review by consideRatio — CHANGES_REQUESTED** [review]
Argh sorry, I forgot that there were breaking changes.

We should cut 2.0.0 instead of 1.4.0, and we should comment the breaking changes briefly.

1. https://github.com/jupyterhub/ltiauthenticator/pull/91
2. https://github.com/jupyterhub/ltiauthenticator/pull/116

**Comment by consideRatio** [comment]
@martinclaus sorry for the back and forth. I see now that the only obviously breaking change was related to LTI13. So that leaves me with the consideration of Python 3.6 and Python 3.7 being removed - is that to be considered breaking?

We have discussed this back and forth in various github repositories. @minrk should we make this 2.0.0 or 1.4.0 based on removing Python 3.6 + 3.7 support?

**Comment by martinclaus** [comment]
Sorry, I have overlooked #116. I interpret the [semantic versioning specs](https://semver.org/#spec-item-8) such that a major version bump is necessary. In particular since even the most recent release of Jupyterhub depends on Python>=3.7. Hence, manually updating `jupyterhub-ltiauthenticator` in an existing environment may break it. Although this should not be an issue when using a package manager, it cannot be assumed that one is used.

**Comment by minrk** [comment]
I think you are free to make it a major version bump if you choose, but it's not required, and I personally wouldn't. There's no harm either way, and I don't have a strong feeling about it. For consistency, it doesn't appear to be standard practice across Python packages to consider an increase in minimum Python version something that prompts a major revision. Nobody's experience is broken by updating the minimum Python version because installers (pip, conda, etc.) will not consider it as an installation candidate for the dropped Python.

**Comment by consideRatio** [comment]
image


Nice!

**Pull request merged**